internal/proto/local/plugin/host_plugin_service.proto

```diff
@@ -0,0 +1,174 @@
+  controller.api.resources.hostcatalogs.v1.HostCatalog new_catalog = 20;
+  
+  // The existing persisted secret data.
+  HostCatalogPersisted persisted = 30;
+}
+
+message OnUpdateCatalogResponse {
+  // The updated secret data to persist encrypted within Boundary.
+  // It's important that this be returned if it existed previously,
+  // as the returned data overwrites the previously existing copy. If
+  // an error is returned, the update of the persisted data is
+  // aborted.
+  HostCatalogPersisted persisted = 10;
+}
+
+message OnDeleteCatalogRequest {
+  // The existing state of the catalog to delete.
+  controller.api.resources.hostcatalogs.v1.HostCatalog catalog = 10;
+  
+  // The existing persisted secret data.
+  HostCatalogPersisted persisted = 20;
+}
+
+message OnDeleteCatalogResponse {}
+
+message OnCreateSetRequest {
+  // The host catalog that the set belongs to.
+  controller.api.resources.hostcatalogs.v1.HostCatalog catalog = 10;
+  
+  // The host set to create.
+  controller.api.resources.hostsets.v1.HostSet set = 20;
+
+  // The persisted data for the host catalog that the set belongs to.
+  HostCatalogPersisted persisted = 30;
+}
+
+message OnCreateSetResponse {}
+
+message OnUpdateSetRequest {
+  // The host catalog that the set belongs to.
+  controller.api.resources.hostcatalogs.v1.HostCatalog catalog = 10;
+  
+  // The existing state of the host set.
+  controller.api.resources.hostsets.v1.HostSet current_set = 20;
+  
+  // The requested new state of the host set.
+  controller.api.resources.hostsets.v1.HostSet new_set = 30;
+
+  // The persisted data for the host catalog that the set belongs to.
+  HostCatalogPersisted persisted = 40;
+}
+
+message OnUpdateSetResponse {}
+
+message OnDeleteSetRequest {
+  // The host catalog that the set belongs to.
+  controller.api.resources.hostcatalogs.v1.HostCatalog catalog = 10;
+  
+  // The host set to delete.
+  controller.api.resources.hostsets.v1.HostSet current_set = 20;
+  
+  // The persisted data for the host catalog that the set belongs to.
+  HostCatalogPersisted persisted = 30;
+}
+
+message OnDeleteSetResponse {}
+
+message ListHostsRequest {
+  // The host catalog that the supplied host sets belong to.
+  controller.api.resources.hostcatalogs.v1.HostCatalog catalog = 10;
+  
+  // The host sets to look up hosts for.
+  repeated controller.api.resources.hostsets.v1.HostSet sets = 20;
+  
+  // The persisted data for the host catalog that the supplied host
+  // sets belong to.
+  HostCatalogPersisted persisted = 30;
+}
+
+message ListHostsResponse {
+  // The hosts to return.
+  repeated ListHostsResponseHost hosts = 10;
+}
+
+message ListHostsResponseHost {
+  // Required. A stable identifier for this host. This field is used
+  // to generate a stable ID for the host within Boundary and is
+  // included in audit logs. It should be set to something unique and
+  // useful, ie: a compute instance ID.
+ 	string external_id = 10;
```

Comment:
There is still some strange spacing happening in this proto file.  Here and elsewhere.

Reply:
Fixed in next commit!